frontend/src/components/TaskTableRated.tsx

```diff
@@ -0,0 +1,226 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import React, { useState } from 'react';
+import { ArrowDownCircle, ArrowUpCircle } from 'react-bootstrap-icons';
+import { Trans } from 'react-i18next';
+import { shallowEqual, useSelector } from 'react-redux';
+import classNames from 'classnames';
+import { TableRatedTaskRow } from './TableRatedTaskRow';
+import { Task, Customer, RoadmapUser, Roadmap } from '../redux/roadmaps/types';
+import { RootState } from '../redux/types';
+import { userInfoSelector } from '../redux/user/selectors';
+import { UserInfo } from '../redux/user/types';
+import { RoleType } from '../../../shared/types/customTypes';
+import {
+  filterTasks,
+  FilterTypes,
+  SortingOrders,
+  SortingTypes,
+  sortTasks,
+  taskAwaitsRatings,
+} from '../utils/TaskUtils';
+import css from './TaskTableRated.module.scss';
+import {
+  allCustomersSelector,
+  chosenRoadmapSelector,
+  roadmapUsersSelector,
+} from '../redux/roadmaps/selectors';
+import { InfoButton } from './forms/InfoButton';
+import { getType } from '../utils/UserUtils';
+
+const classes = classNames.bind(css);
+
+interface TableHeader {
+  label: string;
+  sorting: SortingTypes;
+  textAlign?: 'end' | 'left' | 'center';
+  width?: string;
+}
+
+export const TaskTableRated: React.FC<{
+  tasks: Task[];
+  searchString: string;
+  searchFilter: FilterTypes;
+}> = ({ tasks, searchString, searchFilter }) => {
+  const [sortingType, setSortingType] = useState(SortingTypes.NO_SORT);
+  const [sortingOrder, setSortingOrder] = useState(SortingOrders.ASCENDING);
+  const userInfo = useSelector<RootState, UserInfo | undefined>(
+    userInfoSelector,
+    shallowEqual,
+  );
+
+  const allUsers = useSelector<RootState, RoadmapUser[] | undefined>(
+    roadmapUsersSelector,
+    shallowEqual,
+  );
+  const allCustomers = useSelector<RootState, Customer[] | undefined>(
+    allCustomersSelector,
+    shallowEqual,
+  );
+  const currentRoadmap = useSelector<RootState, Roadmap | undefined>(
+    chosenRoadmapSelector,
+    shallowEqual,
+  );
+
+  const getRenderTaskList: () => Task[] = () => {
+    // Filter, search, sort tasks
+    const filtered = filterTasks(tasks, searchFilter, userInfo?.id);
+    const searched = filtered.filter(
+      (task) =>
+        task.name.toLowerCase().includes(searchString) ||
+        task.description.toLowerCase().includes(searchString),
+    );
+    const sorted = sortTasks(searched, sortingType, sortingOrder);
+
+    return sorted;
+  };
+
+  // Return tasks that don't have ratings from everyone involved in the task - View for product owner
+  const getOwnerTask: () => Task[] = () => {
+    const developers = allUsers?.filter(
+      (user) => user.type === RoleType.Developer,
+    );
```

Comment:
These functions, that are in both TaskTableRated and TaskTableUnrated -files (getOwnerTask, getUnratedTasks?), could be moved to utils-folder

Reply:
True 👍 I realized now that I've used functions like these, such as finding missing devs etc.,  in other files as well that could (and should) be moved to the utils-folder. I think I should make a separate pull-request for that since it will be a bit bigger fix. Did other changes in this pr work well?

Reply:
A separate pull-request for sounds good if a lot needs to be refactored 👍 However, if changes related to this pr are small, they could be added here (others in the separate pr). Up to you!